src/daft-connect/src/convert/expression.rs

```diff
@@ -0,0 +1,120 @@
+            );
+
+            let op = function_name.as_str();
+            match op {
+                ">" | "<" | "<=" | ">=" | "+" | "-" | "*" | "/" => {
+                    let arr: [Expression; 2] = arguments
+                        .try_into()
+                        .map_err(|_| eyre!("Expected 2 arguments"))?;
+                    let [left, right] = arr;
+
+                    let left = convert_expression(left)?;
+                    let right = convert_expression(right)?;
+
+                    let op = match op {
+                        ">" => Operator::Gt,
+                        "<" => Operator::Lt,
+                        "<=" => Operator::LtEq,
+                        ">=" => Operator::GtEq,
+                        "+" => Operator::Plus,
+                        "-" => Operator::Minus,
+                        "*" => Operator::Multiply,
+                        "/" => Operator::FloorDivide, // todo is this what we want?
+                        _ => unreachable!(),
+                    };
+
+                    Ok(DaftExpr::BinaryOp {
+                        left: left.into(),
+                        op,
+                        right: right.into(),
+                    })
+                }
+                other => bail!("Unsupported function name: {other}"),
+            }
+        }
+
+        // Handle other expression types...
+        _ => Err(eyre!("Unsupported expression type")),
+    }
+}
+
+// Helper functions to convert literals, function names, operators etc.
+
+fn convert_literal(lit: expression::Literal) -> Result<daft_dsl::LiteralValue> {
+    let literal_type = lit
+        .literal_type
+        .ok_or_else(|| eyre!("literal_type is None"))?;
+
+    let result = match literal_type {
+        LiteralType::Null(..) => daft_dsl::LiteralValue::Null,
+        LiteralType::Binary(input) => daft_dsl::LiteralValue::Binary(input),
+        LiteralType::Boolean(input) => daft_dsl::LiteralValue::Boolean(input),
+        LiteralType::Byte(input) => daft_dsl::LiteralValue::Int32(input),
+        LiteralType::Short(input) => daft_dsl::LiteralValue::Int32(input),
+        LiteralType::Integer(input) => daft_dsl::LiteralValue::Int32(input),
+        LiteralType::Long(input) => daft_dsl::LiteralValue::Int64(input),
+        LiteralType::Float(input) => daft_dsl::LiteralValue::Float64(f64::from(input)),
+        LiteralType::Double(input) => daft_dsl::LiteralValue::Float64(input),
+        LiteralType::String(input) => daft_dsl::LiteralValue::Utf8(input),
+        LiteralType::Date(input) => daft_dsl::LiteralValue::Date(input),
+        LiteralType::Decimal(_)
+        | LiteralType::Timestamp(_)
+        | LiteralType::TimestampNtz(_)
+        | LiteralType::CalendarInterval(_)
+        | LiteralType::YearMonthInterval(_)
+        | LiteralType::DayTimeInterval(_)
+        | LiteralType::Array(_)
+        | LiteralType::Map(_)
+        | LiteralType::Struct(_) => bail!("unimplemented"),
```

Comment:
i'd change this to match on the wildcard and return the type in the error message. 
```suggestion
        lit => bail!("Literal Value '{lit}' is not yet supported"),
```